server/src/main/java/org/opensearch/search/aggregations/metrics/SumAggregator.java

```diff
@@ -70,25 +70,19 @@ public class SumAggregator extends NumericMetricsAggregator.SingleValue {
         Map<String, Object> metadata
     ) throws IOException {
         super(name, context, parent, metadata);
-        // TODO: stop expecting nulls here
-        this.valuesSource = valuesSourceConfig.hasValues() ? (ValuesSource.Numeric) valuesSourceConfig.getValuesSource() : null;
+        this.valuesSource = (ValuesSource.Numeric) valuesSourceConfig.getValuesSource();
```

Comment:
I don't know the consequence of calling getValuesSource() when hasValues() is false - I'd suggest keeping the original logic here.

Reply:
Yes.  Thanks.  I didn't see that before.